### PR TITLE
Unit Testing the Embedded Examples

### PR DESCRIPTION
**Pull request by joakime**
The /examples/embedded/ code is starting show some bit-rot, so it's time to add testing to them.

I'm implementing a `createServer(int port)` style method for each example that can be both used from the `main(String args[])` method with a static value (like `8080`) and also from junit testing with a port `0` (wildcard) for testing reasons.

Signed-off-by: Joakim Erdfelt <joakim.erdfelt@gmail.com>

### CODE REVIEW

File: examples/embedded/src/main/java/org/eclipse/jetty/embedded/ExampleServer.java [hunk 1]
```diff
@@ -45,6 +45,12 @@ public static void main(String[] args) throws Exception
         handlers.setHandlers(new Handler[]{context, new DefaultHandler()});
         server.setHandler(handlers);
 
+        return server;
+    }
+
+    public static void main(String[] args) throws Exception
+    {
+        Server server = createServer(8080);
```
Comment (gregw) [inline examples/embedded/src/main/java/org/eclipse/jetty/embedded/ExampleServer.java]: Having made the port configurable, you may as well set it from a jetty.port system property that defaults to 8080

Comment (joakime) [reply examples/embedded/src/main/java/org/eclipse/jetty/embedded/ExampleServer.java]: If we did that we'd wind up with ...

``` java

int port = Integer.parseInt(System.getProperty("jetty.port", "8080"));
Server server = createServer(port);
```

Is that really what you want?

Comment (gregw) [reply examples/embedded/src/main/java/org/eclipse/jetty/embedded/ExampleServer.java]: yep - but I'm only +0 on it. if you don't want it don't do it.

Comment (joakime) [reply examples/embedded/src/main/java/org/eclipse/jetty/embedded/ExampleServer.java]: I've gone ahead and implemented the configurable port.

**Pull request merged**